chore(curriculum): adding set 4 of transcripts to DOM lecture block (#58690)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-the-dom-click-events-and-web-apis/6733694805a85d34ced08a9b.md

Watch the lecture video and answer the questions below.

# --transcript--

How do the `setTimeout()` and `setInterval()` methods work?

When building your programs, you'll often want something to happen after a delay, or repeat at regular intervals. That's where the `setTimeout()` and `setInterval()` methods come in handy. Both methods accept two parameters: a function and a delay.

Let's look at each of the methods in more detail so you can start using them in your web projects. 

`setTimeout()` lets you delay an action for a specified time. Here's the basic syntax of the `setTimeout()` method:

```js
setTimeout(functionToRun, delay);
```

> `functionToRun` is the code you want to execute after a specified delay, and `delay` is the time (in milliseconds) to wait before running the function.
> 
> Here's an example:

Reviewer: ```suggestion
`functionToRun` is the code you want to execute after a specified delay, and `delay` is the time (in milliseconds) to wait before running the function. Here's an example:
```

```js
setTimeout(function () {
 console.log("This runs after 3 seconds");
}, 3000);
```

You can also use an arrow function for the callback, like this:

```js
setTimeout(() => {
 console.log("This runs after 3 seconds"); 
}, 3000);
```

> `setInterval()` keeps repeating a function at a set interval. It’s like having an alarm that goes off every few seconds. `setInterval()` is perfect when you want something to happen over and over again, like refreshing data or creating an animation that moves continuously.

Reviewer: ```suggestion
`setInterval()` keeps repeating a function at a set interval. It's like having an alarm that goes off every few seconds. `setInterval()` is perfect when you want something to happen over and over again, like refreshing data or creating an animation that moves continuously.
```

Just like `setTimeout()`, it takes a function and a delay:

```js
setInterval(functionToRun, delay);
```

> `functionToRun` is the code you want to run repeatedly, while `delay` is the time in milliseconds between each execution of that function.
> 
> Here's an example:

Reviewer: ```suggestion
`functionToRun` is the code you want to run repeatedly, while `delay` is the time in milliseconds between each execution of that function. Here's an example:
```
